src/logic_utils/canonical_forms.cpp

```diff
@@ -0,0 +1,157 @@
+#include <algorithm>
+#include <iostream>
+#include <stack>
+#include <string>
+#include <vector>
+using namespace std;
+#include "truth_table_generator.cpp"
+
+// Function to find unique variables in the expression
+vector<Token> unique(vector<Token> expression) {

```

Comment:
Also, remove the unique function from here and put it in the TT generator file and edit the the functions there to use it, and use it here as well, so that we avoid any redundant code. That way `unique` function will be defined only in the TT file, and used there as well as in this file.